Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -454,8 +454,13 @@ class TabTrayController: UIViewController, UITabBarDelegate, UICollectionViewDel
     }
 
     func SELdidClickAddTab() {
-        tabManager?.addTab()
-        presentingViewController!.dismissViewControllerAnimated(true, completion: nil)
+        self.collectionView.performBatchUpdates({ _ in
+            self.tabManager.addTab()
```

Comment:
What's the batch update here for? I know we need to insert a tab in the `didAddTab` callback, but it's not obvious what else we're doing. Maybe add a comment describing the operations that happen under the covers here?